org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/launch/DefaultGradleLaunchConfigurationManager.java

```diff
@@ -74,7 +75,13 @@ private ILaunchConfiguration createLaunchConfiguration(GradleRunConfigurationAtt
             configurationAttributes.apply(launchConfiguration);
 
             // persist the launch configuration and return it
-            return launchConfiguration.doSave();
+            try {
+                return launchConfiguration.doSave();
+            } catch (CoreException ex) {
+                // give a warning if any exception happended during persisting launch configuration file
+                CorePlugin.logger().warn("Could not persisting Gradle launch configuration file", ex);
```

Comment:
❌ Adjust related test coverage: `GradleLaunchConfigurationManagerTest."If can't save attributes then a runtime exception is thrown"`. You'll find several examples in the tests how to mock and verify log messages (hint: search for "`Mock(Logger)`" ).